manager/controlapi/service.go

```diff
@@ -599,8 +600,11 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			}
 
 			curSpec := service.Spec.Copy()
+			curSpecVersion := service.SpecVersion
 			service.Spec = *service.PreviousSpec.Copy()
+			service.SpecVersion = service.PreviousSpecVersion
```

Comment:
Can we put `spec_version` for a service inside of `ServiceSpec`? Then you don't need tracking like this.